src/gui/map/map_editor.cpp

```diff
@@ -3901,6 +3911,318 @@ void MapEditorController::createSymbolWidget(QWidget* parent)
 	}
 }
 
+void MapEditorController::exportOgrClicked()
+{
+	const auto& georef = map->getGeoreferencing();
+
+	// Check if map is georeferenced
+	if (georef.getState() == Georeferencing::Local)
+	{
+		QMessageBox::warning(window, tr("No Georeferencing"), tr("The map is not georeferenced.  Cannot export."));
+		return;
+	}
+
+	// Make sure GDAL can work with the georeferencing info
+	OGRSpatialReference o_spat_ref;
+	if (o_spat_ref.importFromProj4(georef.getProjectedCRSSpec().toLatin1().constData()) != OGRERR_NONE)
+	{
+		QMessageBox::warning(window, tr("Error Exporting"), tr("Failed to properly create the export georeferencing info"));
+		return;
+	}
+
+	// Make sure an object is selected
+	if (map->getNumSelectedObjects() == 0) {
+		QMessageBox::warning(window, tr("Nothing selected"), tr("No objects are selected.  Not exporting anything."));
+		return;
+	}
+
+	QSettings settings;
+	QString export_directory = settings.value(QString::fromLatin1("exportFileDirectory"), QDir::homePath()).toString();
+
+	QString filename = QFileDialog::getSaveFileName(
+						   window,
+						   tr("Export GPX, KML, or SHP file"),
+						   export_directory,
+						   tr("GPS Exchange Format (*.gpx);;Keyhole Markup Language (*.kml);;Shapefile (*.shp)"));
+	if (filename.isEmpty() || filename.isNull())
+		return;
+
+	settings.setValue(QString::fromLatin1("exportFileDirectory"), QFileInfo(filename).canonicalPath());
+
+	// Setup format-specific features
+	const char *psz_driver_name;
+	bool gpx_quirks = false;
+	if (filename.endsWith(QLatin1String(".gpx"), Qt::CaseInsensitive))
+	{
+		psz_driver_name = "GPX";
+		gpx_quirks = true;
+	}
+	else if (filename.endsWith(QLatin1String(".kml"), Qt::CaseInsensitive))
+	{
+		psz_driver_name = "KML";
+	}
+	else
+		psz_driver_name = "ESRI Shapefile";
+
+	// GPX driver can't automatically overwrite files
+	if (gpx_quirks)
+	{
+		QFile file(filename);
+		if (file.exists())
+			file.remove();
+	}
+
+	GDALAllRegister();
```

Comment:
GDAL 2.0. We don't want to call this every time. This is done in `GdalManager` on first use.